tests/conftest.py

```diff
@@ -134,3 +140,180 @@ def openlabel_v1_vcd_incompatible_data(request):
             )
 
     return openlabel_v1_vcd_incompatible_data
+
+@pytest.fixture
+def annotation_compare_methods() -> t.Dict[str, t.Callable]:
+    methods = {}
+
+    def compare_bbox(annotation: raillabel.format.Bbox, ground_truth: dict) -> bool:
+
+        assert type(annotation) == raillabel.format.Bbox
+        assert annotation.uid == ground_truth["uid"]
+        assert annotation.name == ground_truth["name"]
+        assert annotation.pos.x == ground_truth["val"][0]
+        assert annotation.pos.y == ground_truth["val"][1]
+        assert annotation.size.x == ground_truth["val"][2]
+        assert annotation.size.y == ground_truth["val"][3]
+
+        if "coordinate_system" in ground_truth:
+            assert annotation.sensor.uid == ground_truth["coordinate_system"]
+
+        if "attributes" in ground_truth:
+
+            accumulative_attributes = []
+            for attr_type in ground_truth["attributes"].values():
+                accumulative_attributes.extend(attr_type)
+
+            assert len(annotation.attributes) == len(accumulative_attributes)
+            for attribute in accumulative_attributes:
+                assert attribute["name"] in annotation.attributes
+                assert annotation.attributes[attribute["name"]] == attribute["val"]
+
+    methods["bbox"] = compare_bbox
+
+
+    def compare_cuboid(annotation: raillabel.format.Cuboid, ground_truth: dict) -> bool:
+
+        assert type(annotation) == raillabel.format.Cuboid
+        assert annotation.uid == ground_truth["uid"]
+        assert annotation.name == ground_truth["name"]
+        assert annotation.pos.x == ground_truth["val"][0]
+        assert annotation.pos.y == ground_truth["val"][1]
+        assert annotation.pos.z == ground_truth["val"][2]
+        assert annotation.quat.x == ground_truth["val"][3]
+        assert annotation.quat.y == ground_truth["val"][4]
+        assert annotation.quat.z == ground_truth["val"][5]
+        assert annotation.quat.w == ground_truth["val"][6]
+        assert annotation.size.x == ground_truth["val"][7]
+        assert annotation.size.y == ground_truth["val"][8]
+        assert annotation.size.z == ground_truth["val"][9]
```

Comment:
Something like 
```python
for annotation_val, index in enumerate([annotation.pos.x, annotation.pos.y, annotation.pos.z ...]):
    assert annotation_val == ground_truth["val"][index]
```
is imho less prone to index errors, in case this code gets ever touched again

Reply:
Implemented